test/bootboot_test.rb

```diff
@@ -240,6 +240,19 @@ def test_bundle_install_with_different_ruby_updating_gemfile_next_lock_succeeds
     end
   end
 
+  def test_bundle_install_without_specific_ruby_version_succeeds
+    write_gemfile do |file, _dir|
+      requirement = "~> #{RUBY_VERSION.rpartition('.')[0]}.0" # transforms 2.7.7 into ~> 2.7.0
```

Comment:
It's very much required, as otherwise the bug isn't reproduced at all. Regardless of the actual versions available, the requirement being different means that there **can** be multiple versions.

I've reworded the comment to make the intent more clearer.

Reply:
Sorry I should have pointed that I was asking about the `rpartition` and substitution part, I do agree that turning the requirement into `~>` is necessary but  I believe the test case should also work if we do `requirement = "~> #{RUBY_VERSION}"` so we transform `2.7.7` into `~> 2.7.7` without nullifying the patch version as `~> 2.7.7` is not an exact version as well